lexer.go

```diff
@@ -105,9 +106,13 @@ type Lexer interface {
 	//
 	// Lexers may ignore this if they implement their own analysers.
 	SetAnalyser(analyser func(text string) float32) Lexer
-	// AnalyseText scores how likely a fragment of text is to match
-	// this lexer, between 0.0 and 1.0. A value of 1 indicates high confidence.
+	// AnalyseText scores how likely a fragment of text is to match this lexer,
+	// between 0.0 and 1.0. A value of 1 indicates high confidence.
 	AnalyseText(text string) float32
+	// SetRegexTimeout sets the timeout after which the regex matcher gives up.
+	// The default timeout is 250 ms. To disable the timeout, set it to zero.
+	// This currently only affects RegexLexer.
+	SetRegexTimeout(timeout time.Duration)
```

Comment:
The idea of the `Lexer` interface is that it can be used by other lexers that are non-regex specific, so putting this method in here is coupling the interface too tightly to the RegexLexer implementation. That said, I think just renaming it `SetTimeout()` is general enough to be reasonable.

Reply:
Done!